packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -0,0 +1,205 @@
+        ' as deferred, allowing for lazy loading of each locale in Flutter web.\n'
+        '\n'
+        'This can reduce a web app’s initial startup time by decreasing the '
+        'size of the JavaScript bundle. When this flag is set to true, the '
+        'messages for a particular locale are only downloaded and loaded by the '
+        'Flutter app as they are needed. For projects with a lot of different '
+        'locales and many localization strings, it can be an performance '
+        'improvement to have deferred loading. For projects with a small number '
+        'of locales, the difference is negligible, and might slow down the start '
+        'up compared to bundling the localizations with the rest of the '
+        'application.\n\n'
+        'Note that this flag does not affect other platforms such as mobile or '
+        'desktop.',
+    );
+    argParser.addOption(
+      'gen-inputs-and-outputs-list',
+      valueHelp: 'path-to-output-directory',
+      help: 'When specified, the tool generates a JSON file containing the '
+        'tool\'s inputs and outputs named gen_l10n_inputs_and_outputs.json.'
+        '\n\n'
+        'This can be useful for keeping track of which files of the Flutter '
+        'project were used when generating the latest set of localizations. '
+        'For example, the Flutter tool\'s build system uses this file to '
+        'keep track of when to call gen_l10n during hot reload.\n\n'
+        'The value of this option is the directory where the JSON file will be '
+        'generated.'
+        '\n\n'
+        'When null, the JSON file will not be generated.'
+    );
+    argParser.addFlag(
+      'synthetic-package',
+      defaultsTo: true,
+      help: 'Determines whether or not the generated output files will be '
+        'generated as a synthetic package or at a specified directory in '
+        'the Flutter project.'
+        '\n\n'
+        'This flag is set to true by default.'
+        '\n\n'
+        'When synthetic-package is set to false, it will generate the '
+        'localizations files in the directory specified by arb-dir by default. '
+        '\n\n'
+        'If output-dir is specified, files will be generated there.',
+    );
+    argParser.addOption(
+      'project-dir',
+      valueHelp: 'absolute/path/to/flutter/project',
+      help: 'When specified, the tool uses the path passed into this option '
+        'as the directory of the root Flutter project.'
+        '\n\n'
+        'When null, the relative path to the present working directory will be used.'
+    );
+  }
+
+  FileSystem _fileSystem;
+
+  @override
+  String get description => 'Generate localizations for the Flutter project.';
+
+  @override
+  String get name => 'generate_localizations';
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+
+    precacheLanguageAndRegionTags();
+
+    final String inputPathString = argResults['arb-dir'] as String;
```

Comment:
there are some helpers like `stringArg`, `stringsArg`, et cetera that avoid the casting